contrib/datadog/tracing/interceptor.go

```diff
@@ -242,8 +247,21 @@ func (t *tracerSpan) ForeachBaggageItem(handler func(k string, v string) bool) {
 
 func (t *tracerSpan) Finish(options *interceptor.TracerFinishSpanOptions) {
 	var opts []tracer.FinishOption
-	if err := options.Error; err != nil && !workflow.IsContinueAsNewError(err) {
+
+	err := options.Error
+	isErrEligible := err != nil && !workflow.IsContinueAsNewError(err) && t.shouldErrBeTraced(err)
+
+	if isErrEligible {
 		opts = append(opts, tracer.WithError(err))
 	}
+
 	t.Span.Finish(opts...)
 }
+
+func (t *tracerSpan) shouldErrBeTraced(err error) bool {
```

Comment:
Can just inline this, no need for a separate method (but no biggie)

Reply:
Addressed